source/sema.h

```diff
@@ -1922,6 +1926,18 @@ class sema
 
     auto end(compound_statement_node const& n, int) -> void
     {
+        //  Pop an implicit 'else' branch.
+        if (auto s = std::find_if(
+                         symbols.rbegin(),
+                         symbols.rend(),
+                         [=](symbol s) {
+                             return s.depth == scope_depth - 1;
+                         });
+            s == symbols.rend()
+            || std::get_if<symbol::selection>(&s->sym)
+            ) {
+            --scope_depth;
+        }
```

Comment:
Please, confirm that commit aef69a53818311dd940be5008d838a0280fc6dfe is correct and is the right direction.
I suspect it is, and that this 'pop' action should be factored out into a function
and called from the various `end` function of all the statements that can contain a selection statement.

Reply:
All braces are *compound-statement*s, so it seems like there's nothing to factor out.

Reply:
Superseded by commit e9cc033b36e47de783a3c4ae12174d66fc592fa6.